src/equations/shallow_water_multilayer_2d.jl

```diff
@@ -0,0 +1,828 @@
+          (0.1sin(t + x[1] * ω) * ω + 0.1cos(t + x[1] * ω) * ω) +
+          10.0(0.5 - 0.1cos(t + x[2] * ω) + 0.1sin(t + x[1] * ω) +
+               0.1sin(t + x[2] * ω) - 0.1cos(t + x[1] * ω)) *
+          (-0.1sin(t + x[1] * ω) * ω +
+           0.9(-0.1sin(t + x[1] * ω) * ω - 0.1cos(t + x[1] * ω) * ω))
+
+    du6 = 0.8(-0.1sin(t + x[1] * ω) - 0.1sin(t + x[2] * ω)) +
+          0.8(0.1sin(x[2] * ω) * ω - 0.1sin(t + x[2] * ω) * ω) +
+          0.8^2 * (0.1sin(x[1] * ω) * ω - 0.1sin(t + x[1] * ω) * ω) +
+          10.0(0.5 - 0.1cos(x[1] * ω) + 0.1cos(t + x[2] * ω) - 0.1cos(x[2] * ω) +
+               0.1cos(t + x[1] * ω)) *
+          (0.1sin(x[1] * ω) * ω - 0.1sin(t + x[1] * ω) * ω) +
+          10.0(0.5 - 0.1cos(x[1] * ω) + 0.1cos(t + x[2] * ω) - 0.1cos(x[2] * ω) +
+               0.1cos(t + x[1] * ω)) * (-0.1sin(x[1] * ω) * ω +
+           0.9 / 1.1 * (-0.1sin(t + x[1] * ω) * ω - 0.1cos(t + x[1] * ω) * ω) +
+           1.0 / 1.1 * (0.1sin(t + x[1] * ω) * ω + 0.1cos(t + x[1] * ω) * ω))
+
+    du7 = -0.1cos(t + x[2] * ω) - 0.1sin(t + x[1] * ω) - 0.1sin(t + x[2] * ω) -
+          0.1cos(t + x[1] * ω) - 0.1cos(t + x[2] * ω) * ω +
+          0.8(-0.1sin(t + x[1] * ω) * ω - 0.1cos(t + x[1] * ω) * ω) -
+          0.1sin(t + x[2] * ω) * ω +
+          10.0(2.0 + 0.1cos(t + x[2] * ω) - 0.1sin(t + x[1] * ω) -
+               0.1sin(t + x[2] * ω) + 0.1cos(t + x[1] * ω)) *
+          (-0.1cos(t + x[2] * ω) * ω - 0.1sin(t + x[2] * ω) * ω) +
+          (2.0 + 0.1cos(t + x[2] * ω) - 0.1sin(t + x[1] * ω) - 0.1sin(t + x[2] * ω) +
+           0.1cos(t + x[1] * ω)) * cos(t + x[2] * ω) * ω
+
+    du8 = 0.1cos(t + x[2] * ω) + 0.1sin(t + x[1] * ω) + 0.1sin(t + x[2] * ω) +
+          0.1cos(t + x[1] * ω) + 0.1cos(t + x[2] * ω) * ω +
+          0.8(0.1sin(t + x[1] * ω) * ω + 0.1cos(t + x[1] * ω) * ω) +
+          0.1sin(t + x[2] * ω) * ω +
+          10.0(0.5 - 0.1cos(t + x[2] * ω) + 0.1sin(t + x[1] * ω) +
+               0.1sin(t + x[2] * ω) - 0.1cos(t + x[1] * ω)) *
+          (0.9(-0.1cos(t + x[2] * ω) * ω - 0.1sin(t + x[2] * ω) * ω) -
+           0.1sin(t + x[2] * ω) * ω) +
+          10.0(0.5 - 0.1cos(t + x[2] * ω) + 0.1sin(t + x[1] * ω) +
+               0.1sin(t + x[2] * ω) - 0.1cos(t + x[1] * ω)) *
+          (0.1cos(t + x[2] * ω) * ω + 0.1sin(t + x[2] * ω) * ω)
+
+    du9 = -0.1sin(t + x[1] * ω) - 0.1sin(t + x[2] * ω) + 0.1sin(x[2] * ω) * ω +
+          0.8(0.1sin(x[1] * ω) * ω - 0.1sin(t + x[1] * ω) * ω) -
+          0.1sin(t + x[2] * ω) * ω +
+          10.0(0.5 - 0.1cos(x[1] * ω) + 0.1cos(t + x[2] * ω) - 0.1cos(x[2] * ω) +
+               0.1cos(t + x[1] * ω)) *
+          (0.9 / 1.1 * (-0.1cos(t + x[2] * ω) * ω - 0.1sin(t + x[2] * ω) * ω) +
+           1.0 / 1.1 * (0.1cos(t + x[2] * ω) * ω + 0.1sin(t + x[2] * ω) * ω) -
+           0.1sin(x[2] * ω) * ω) +
+          10.0(0.5 - 0.1cos(x[1] * ω) + 0.1cos(t + x[2] * ω) - 0.1cos(x[2] * ω) +
+               0.1cos(t + x[1] * ω)) * (0.1sin(x[2] * ω) * ω - 0.1sin(t + x[2] * ω) * ω)
```

Comment:
```suggestion
    du1 = (-0.1 * cos(t + x[2] * ω) - 0.1 * sin(t + x[1] * ω) - 0.1 * sin(t + x[2] * ω) -
          0.1 * cos(t + x[1] * ω) - 0.1 * cos(t + x[2] * ω) * ω +
          0.8 * (-0.1 * sin(t + x[1] * ω) * ω - 0.1 * cos(t + x[1] * ω) * ω) -
          0.1 * sin(t + x[2] * ω) * ω)

    du2 = (0.1 * cos(t + x[2] * ω) + 0.1 * sin(t + x[1] * ω) + 0.1 * sin(t + x[2] * ω) +
          0.1 * cos(t + x[1] * ω) + 0.1 * cos(t + x[2] * ω) * ω +
          0.8 * (0.1 * sin(t + x[1] * ω) * ω + 0.1 * cos(t + x[1] * ω) * ω) +
          0.1 * sin(t + x[2] * ω) * ω)

    du3 = (-0.1 * sin(t + x[1] * ω) - 0.1 * sin(t + x[2] * ω) + 0.1 * sin(x[2] * ω) * ω +
          0.8 * (0.1 * sin(x[1] * ω) * ω - 0.1 * sin(t + x[1] * ω) * ω) -
          0.1 * sin(t + x[2] * ω) * ω)

    du4 = (0.8 * (-0.1 * cos(t + x[2] * ω) - 0.1 * sin(t + x[1] * ω) - 0.1 * sin(t + x[2] * ω) -
              0.1 * cos(t + x[1] * ω)) +
          0.8 * (-0.1 * cos(t + x[2] * ω) * ω - 0.1 * sin(t + x[2] * ω) * ω) +
          0.8^2 * (-0.1 * sin(t + x[1] * ω) * ω - 0.1 * cos(t + x[1] * ω) * ω) +
          10.0 * (2.0 + 0.1 * cos(t + x[2] * ω) - 0.1 * sin(t + x[1] * ω) -
               0.1 * sin(t + x[2] * ω) + 0.1 * cos(t + x[1] * ω)) *
          (-0.1 * sin(t + x[1] * ω) * ω - 0.1 * cos(t + x[1] * ω) * ω) +
          (2.0 + 0.1 * cos(t + x[2] * ω) - 0.1 * sin(t + x[1] * ω) - 0.1 * sin(t + x[2] * ω) +
           0.1 * cos(t + x[1] * ω)) * cos(t + x[1] * ω) * ω)

    du5 = (0.8 * (0.1 * cos(t + x[2] * ω) + 0.1 * sin(t + x[1] * ω) + 0.1 * sin(t + x[2] * ω) +
              0.1 * cos(t + x[1] * ω)) +
          0.8 * (0.1 * cos(t + x[2] * ω) * ω + 0.1 * sin(t + x[2] * ω) * ω) +
          0.8^2 * (0.1 * sin(t + x[1] * ω) * ω + 0.1 * cos(t + x[1] * ω) * ω) +
          10.0 * (0.5 - 0.1 * cos(t + x[2] * ω) + 0.1 * sin(t + x[1] * ω) +
               0.1 * sin(t + x[2] * ω) - 0.1 * cos(t + x[1] * ω)) *
          (0.1 * sin(t + x[1] * ω) * ω + 0.1 * cos(t + x[1] * ω) * ω) +
          10.0 * (0.5 - 0.1 * cos(t + x[2] * ω) + 0.1 * sin(t + x[1] * ω) +
               0.1 * sin(t + x[2] * ω) - 0.1 * cos(t + x[1] * ω)) *
          (-0.1 * sin(t + x[1] * ω) * ω +
           0.9 * (-0.1 * sin(t + x[1] * ω) * ω - 0.1 * cos(t + x[1] * ω) * ω)))

    du6 = (0.8 * (-0.1 * sin(t + x[1] * ω) - 0.1 * sin(t + x[2] * ω)) +
          0.8 * (0.1 * sin(x[2] * ω) * ω - 0.1 * sin(t + x[2] * ω) * ω) +
          0.8^2 * (0.1 * sin(x[1] * ω) * ω - 0.1 * sin(t + x[1] * ω) * ω) +
          10.0 * (0.5 - 0.1 * cos(x[1] * ω) + 0.1 * cos(t + x[2] * ω) - 0.1 * cos(x[2] * ω) +
               0.1 * cos(t + x[1] * ω)) *
          (0.1 * sin(x[1] * ω) * ω - 0.1 * sin(t + x[1] * ω) * ω) +
          10.0 * (0.5 - 0.1 * cos(x[1] * ω) + 0.1 * cos(t + x[2] * ω) - 0.1 * cos(x[2] * ω) +
               0.1 * cos(t + x[1] * ω)) * (-0.1 * sin(x[1] * ω) * ω +
           0.9 / 1.1 * (-0.1 * sin(t + x[1] * ω) * ω - 0.1 * cos(t + x[1] * ω) * ω) +
           1.0 / 1.1 * (0.1 * sin(t + x[1] * ω) * ω + 0.1 * cos(t + x[1] * ω) * ω)))

    du7 = (-0.1 * cos(t + x[2] * ω) - 0.1 * sin(t + x[1] * ω) - 0.1 * sin(t + x[2] * ω) -
          0.1 * cos(t + x[1] * ω) - 0.1 * cos(t + x[2] * ω) * ω +
          0.8 * (-0.1 * sin(t + x[1] * ω) * ω - 0.1 * cos(t + x[1] * ω) * ω) -
          0.1 * sin(t + x[2] * ω) * ω +
          10.0 * (2.0 + 0.1 * cos(t + x[2] * ω) - 0.1 * sin(t + x[1] * ω) -
               0.1 * sin(t + x[2] * ω) + 0.1 * cos(t + x[1] * ω)) *
          (-0.1 * cos(t + x[2] * ω) * ω - 0.1 * sin(t + x[2] * ω) * ω) +
          (2.0 + 0.1 * cos(t + x[2] * ω) - 0.1 * sin(t + x[1] * ω) - 0.1 * sin(t + x[2] * ω) +
           0.1 * cos(t + x[1] * ω)) * cos(t + x[2] * ω) * ω)

    du8 = (0.1 * cos(t + x[2] * ω) + 0.1 * sin(t + x[1] * ω) + 0.1 * sin(t + x[2] * ω) +
          0.1 * cos(t + x[1] * ω) + 0.1 * cos(t + x[2] * ω) * ω +
          0.8 * (0.1 * sin(t + x[1] * ω) * ω + 0.1 * cos(t + x[1] * ω) * ω) +
          0.1 * sin(t + x[2] * ω) * ω +
          10.0 * (0.5 - 0.1 * cos(t + x[2] * ω) + 0.1 * sin(t + x[1] * ω) +
               0.1 * sin(t + x[2] * ω) - 0.1 * cos(t + x[1] * ω)) *
          (0.9 * (-0.1 * cos(t + x[2] * ω) * ω - 0.1 * sin(t + x[2] * ω) * ω) -
           0.1 * sin(t + x[2] * ω) * ω) +
          10.0 * (0.5 - 0.1 * cos(t + x[2] * ω) + 0.1 * sin(t + x[1] * ω) +
               0.1 * sin(t + x[2] * ω) - 0.1 * cos(t + x[1] * ω)) *
          (0.1 * cos(t + x[2] * ω) * ω + 0.1 * sin(t + x[2] * ω) * ω))

    du9 = (-0.1 * sin(t + x[1] * ω) - 0.1 * sin(t + x[2] * ω) + 0.1 * sin(x[2] * ω) * ω +
          0.8 * (0.1 * sin(x[1] * ω) * ω - 0.1 * sin(t + x[1] * ω) * ω) -
          0.1 * sin(t + x[2] * ω) * ω +
          10.0 * (0.5 - 0.1 * cos(x[1] * ω) + 0.1 * cos(t + x[2] * ω) - 0.1 * cos(x[2] * ω) +
               0.1 * cos(t + x[1] * ω)) *
          (0.9 / 1.1 * (-0.1 * cos(t + x[2] * ω) * ω - 0.1 * sin(t + x[2] * ω) * ω) +
           1.0 / 1.1 * (0.1 * cos(t + x[2] * ω) * ω + 0.1 * sin(t + x[2] * ω) * ω) -
           0.1 * sin(x[2] * ω) * ω) +
          10.0 * (0.5 - 0.1 * cos(x[1] * ω) + 0.1 * cos(t + x[2] * ω) - 0.1 * cos(x[2] * ω) +
               0.1 * cos(t + x[1] * ω)) * (0.1 * sin(x[2] * ω) * ω - 0.1 * sin(t + x[2] * ω) * ω))
```
It is true that it blows up the lines a little bit adding the multiplication symbols, but I like that it avoids any ambiguity. I also added parentheses around each longer expression because I have burned in the past if they are left out.

Reply:
Thanks for preparing the formatting. I am also fine with having the multiplication symbols here. Done in 6fc27e2dab0532d44c931e57f2ba2fa6fb56f36b